### PR TITLE
Module 1:store form inputs with local storage

### PR DESCRIPTION
**Pull request by derrick1451**
### Local storage 

- We updated our page and implemented local storage on it by

1. Using javascript and the dom 
2. Editng the HTML file
3. Make a few changes to the CSS

### CODE REVIEW

**Comment by derrick1451** [comment]
Screenshot from 2023-04-19 16-44-05

File: script.js [hunk 1]
```diff
@@ -150,4 +150,19 @@ form.addEventListener('submit', ((e) => {
...
+};
+const formItems = document.querySelectorAll('.form-container');
+const formData = Array.from(formItems);
+formData.forEach((form) => {
+  form.addEventListener('input', (e) => {
+    const letter = e.target.id;
+    data[letter] = e.target.value;
+    const dataMarker = JSON.stringify(data);
+    localStorage.setItem('data', dataMarker);
+  });
```
Comment (AlphaNtihinduka) [inline script.js:171]: - [ ] I can see the form information is stored in the local storage when the user inputs the data into the form. Which is a very great achievement 👏. I can say we are very close to completing the form data preserve. If you noticed, you are not having the part that can help you to retrieve data from local storage. Therefore I suggest that you modify your code to make sure that the contact form information is persistent to page reloads ☺️.

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @derrick1451 🙋‍♂️
Good job working on preserving form data in the local storage. You have done the design to the best expectation and incredibly well. 👏  

You still need to implement few on the project but you are almost there!!! 🚶: 

TO HIGHLIGHT:
- Your project is professional ✅
- Professional README file :100: 
- Your PR is professional ✅
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

# Required Changes :recycle: 
- Kindly consider the inline review comments.
- Additionally, I have noticed that your Readme has a broken MIT link. Kindly ensure to fix this so that other developers are able to access your license.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

**Comment by derrick1451** [comment]
Could you please through more light on the issue cos am lost with the
review
Do you mean using the getitem or

**Review by aamirkhan2478 — APPROVED** [review]
Hi @derrick1451 , 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly
✅ No linter error
✅ Good commit messages
✅ Good descriptive PR

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AlphaNtihinduka** [comment]
Dear @derrick1451 I am sorry that I wasn't able to read your comment as earlier as it could be possible. However, what you thought about was the requirement I was suggesting for. Please incase you have a question, you should remember to tag the reviewer so that he/she can get the notification for quick response. I am happy that your project was approved and congratulations 👍.
HAPPY CODING.

**Pull request merged**